### PR TITLE
Add tests for special characters in beakerlib phase names

### PR DESCRIPTION
**Pull request by seberm**
**Do not merge this PR until the [beakerlib changes](https://github.com/beakerlib/beakerlib/pull/184) are released.**

This PR adds tests for special characters in beakerlib phase names.  It checks the special characters are correctly handled by beakerlib & tmt and propagated into the internal structures (esp. `results.yaml`, `tmt-report-results.yaml`).

Related to:
- https://github.com/teemtee/tmt/issues/3333
- https://github.com/beakerlib/beakerlib/pull/184

Pull Request Checklist

* [x] extend the test coverage

### CODE REVIEW

**Comment by seberm** [comment]
Hello @psss, this one should be ready for review. I've modified the tests to always check the values of the concrete subresult extracted using the `yq`.

The problems in the CI (in `Fedora-{41,rawhide}`) seems unrelated to my changes.

**Review by happz — APPROVED** [review]

**Review by psss — APPROVED** [review]
Looks good, thanks for improving the test coverage!

File: tests/execute/result/special/beaker-phases-special-names.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+. /usr/share/beakerlib/beakerlib.sh || exit 1
+
+
+rlJournalStart
+    for i in \
+        # Ensure the regression from #teemtee/tmt/3333 is covered
+        /sbin/ldconfig \
+        /usr/sbin/ldconfig \
+
+        # Cover other special characters in the phase name
```
[the remark this answers is not in the document]
Comment (psss) [reply tests/execute/result/special/beaker-phases-special-names.sh]: This seems to be sufficient. Thanks!

**Comment by psss** [comment]
Red tests are irrelevant failures, merging.

**Pull request merged**